modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -193,11 +193,11 @@ class ReplCalculator(
         if (queue.nonEmpty) {
           //queue could be empty if InterruptRequest was asked!
           log.debug("Dequeuing execute request current size: " + queue.size)
-          queue = queue.dequeue._2
-          queue.headOption foreach { case (ref, er) =>
-            log.debug("About to execute request from the queue")
-            execute(ref, er)
-          }
+          val (executeRequest, newQueue) = queue.dequeue
+          queue = newQueue
+          val (ref, er) = executeRequest
```

Comment:
is there any use of keeping the currently executing cell as the first element in the queue?
if needed, maybe better to have a separate variable or even a queue for currently executing items (e.g. I didn't expect the queue to keep the currently executing item).